packages/itwin/grouping-mapping-widget/src/widget/components/GroupingMapping.tsx

```diff
@@ -7,21 +7,51 @@ import { Mappings } from "./Mapping";
 import "./GroupingMapping.scss";
 import type { AccessToken } from "@itwin/core-bentley";
 import { IModelApp } from "@itwin/core-frontend";
+import type { IMappingClient } from "../IMappingClient";
+import { REPORTING_BASE_PATH, ReportingClient } from "@itwin/insights-client";
 
 export interface Api {
   accessToken: AccessToken;
-  prefix?: "" | "dev" | "qa";
+  prefix?: ClientPrefix;
 }
 
-export const ApiContext = createContext<Api>({ accessToken: "" });
+const prefixUrl = (baseUrl?: string, prefix?: string) => {
+  if (prefix && baseUrl) {
+    return baseUrl.replace("api.bentley.com", `${prefix}-api.bentley.com`);
+  }
+  return baseUrl;
+};
+
+export const createDefaultMappingClient = (prefix?: ClientPrefix): IMappingClient => {
+  const url = prefixUrl(REPORTING_BASE_PATH, prefix);
+  return new ReportingClient(url);
+};
+
+export const ApiContext = createContext<Api>({ accessToken: "", prefix: undefined });
+
+export const MappingClientContext = createContext<IMappingClient>(createDefaultMappingClient());
+
+export type ClientPrefix = "" | "dev" | "qa" | undefined;
 
 interface GroupingMappingProps {
   accessToken?: AccessToken;
-  prefix?: "" | "dev" | "qa";
+  /**
+   * Used for iTwin and iModel APIs.
+   * Also used for Mapping API if a custom {@link client} is not provided.
+   */
+  prefix?: ClientPrefix;
+  /**
+   * A custom implementation of MappingClient.
+   */
+  client?: IMappingClient;
 }
 
-const GroupingMapping = ({ accessToken, prefix }: GroupingMappingProps) => {
+const GroupingMapping = (props: GroupingMappingProps) => {
   const [currentAccessToken, setCurrentAccessToken] = useState<string>("");
+  const [mappingClient, setMappingClient] = useState<IMappingClient>(createDefaultMappingClient());
+
+  const { accessToken, prefix } = props;
```

Comment:
Why not destructure this in the parameters?

Reply:
Destructured.